src/linux/transaction.rs

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use platform::monitor::Monitor;
+use runloop::RunLoop;
+use std::ffi::OsString;
+use std::io;
+use util::{io_err, OnceCallback};
+
+pub struct Transaction {
+    // Handle to the thread loop.
+    thread: Option<RunLoop>,
+}
+
+impl Transaction {
```

Comment:
Feels like this could have a common interface with `macos/transaction.rs`, but perhaps that should be a follow-on after whatever happens to Windows.

Reply:
Indeed, I've been thinking about this too. There are tiny differences, e.g. the parameters for `Device::new()` and whether we call `dev.is_u2f()` or not but we can work around that. I agree that we should do this after converting Windows.